diff --git a/src/sfloader/fluid_sfont.c b/src/sfloader/fluid_sfont.c
index 7f61d63f..f37513ca 100644
--- a/src/sfloader/fluid_sfont.c
+++ b/src/sfloader/fluid_sfont.c
@@ -21,39 +21,27 @@
 #include "fluid_sys.h"
 #include "fluid_mod.h"
 
+void *default_fopen(const char *path);
+int default_fclose(void *handle);
+fluid_long_long_t default_ftell(void *handle);
+int safe_fread(void *buf, fluid_long_long_t count, void *handle);
+int safe_fseek(void *handle, fluid_long_long_t ofs, int whence);
 
-void *default_fopen(const char *path)
-{
-    const char* msg;
-    FILE* handle = fluid_file_open(path, &msg);
 
-    if(handle == NULL)
-    {
-        FLUID_LOG(FLUID_ERR, "fluid_sfloader_load(): Failed to open '%s': %s", path, msg);
-    }
+/* mt32-pi: default_fopen provided externally */
 
-    return handle;
-}
 
-int default_fclose(void *handle)
-{
-    return FLUID_FCLOSE((FILE *)handle) == 0 ? FLUID_OK : FLUID_FAILED;
-}
+/* mt32-pi: default_fclose provided externally */
 
-fluid_long_long_t default_ftell(void *handle)
-{
-    return FLUID_FTELL((FILE *)handle);
-}
 
-int safe_fread(void *buf, fluid_long_long_t count, void *handle)
-{
-    return fluid_file_read(buf, count, handle);
-}
+/* mt32-pi: default_ftell provided externally */
+
+
+/* mt32-pi: safe_fread provided externally */
+
+
+/* mt32-pi: safe_fseek provided externally */
 
-int safe_fseek(void *handle, fluid_long_long_t ofs, int whence)
-{
-    return fluid_file_seek(handle, ofs, whence);
-}
 
 /**
  * Creates a new SoundFont loader.
diff --git a/src/utils/fluid_sys.c b/src/utils/fluid_sys.c
index 91d24e0c..60d1408d 100644
--- a/src/utils/fluid_sys.c
+++ b/src/utils/fluid_sys.c
@@ -225,28 +225,8 @@ fluid_log(int level, const char *fmt, ...)
     return FLUID_FAILED;
 }
 
-void* fluid_alloc(size_t len)
-{
-    void* ptr = malloc(len);
+/* mt32-pi: fluid_alloc provided externally */
 
-#if defined(DEBUG) && !defined(_MSC_VER)
-    // garbage initialize allocated memory for debug builds to ease reproducing
-    // bugs like 44453ff23281b3318abbe432fda90888c373022b .
-    //
-    // MSVC++ already garbage initializes allocated memory by itself (debug-heap).
-    //
-    // 0xCC because
-    // * it makes pointers reliably crash when dereferencing them,
-    // * floating points are still some valid but insanely huge negative number, and
-    // * if for whatever reason this allocated memory is executed, it'll trigger
-    //   INT3 (...at least on x86)
-    if(ptr != NULL)
-    {
-        memset(ptr, 0xCC, len);
-    }
-#endif
-    return ptr;
-}
 
 /**
  * Open a file with a UTF-8 string, even in Windows
@@ -305,23 +285,8 @@ error_recovery:
 #endif
 }
 
-/**
- * Wrapper for free() that satisfies at least C90 requirements.
- *
- * @param ptr Pointer to memory region that should be freed
- *
- * @note Only use this function when the API documentation explicitly says so. Otherwise use
- * adequate \c delete_fluid_* functions.
- *
- * @warning Calling C-std lib <code>free()</code> on memory that is advised to be freed with fluid_free() results in undefined behaviour!
- * (cf.: "Potential Errors Passing CRT Objects Across DLL Boundaries" found in MS Docs)
- *
- * @since 2.0.7
- */
-void fluid_free(void* ptr)
-{
-    free(ptr);
-}
+/* mt32-pi: fluid_free provided externally */
+
 
 /**
  * An improved strtok, still trashes the input string, but is portable and
@@ -1640,38 +1605,8 @@ void delete_fluid_server_socket(fluid_server_socket_t *server_socket)
 
 #endif // NETWORK_SUPPORT
 
-FILE* fluid_file_open(const char* path, const char** errMsg)
-{
-    static const char ErrExist[] = "File does not exist.";
-    static const char ErrRegular[] = "File is not regular, refusing to open it.";
-    static const char ErrNull[] = "File does not exists or insufficient permissions to open it.";
-    
-    FILE* handle = NULL;
-    
-    if(!fluid_file_test(path, FLUID_FILE_TEST_EXISTS))
-    {
-        if(errMsg != NULL)
-        {
-            *errMsg = ErrExist;
-        }
-    }
-    else if(!fluid_file_test(path, FLUID_FILE_TEST_IS_REGULAR))
-    {
-        if(errMsg != NULL)
-        {
-            *errMsg = ErrRegular;
-        }
-    }
-    else if((handle = FLUID_FOPEN(path, "rb")) == NULL)
-    {
-        if(errMsg != NULL)
-        {
-            *errMsg = ErrNull;
-        }
-    }
-    
-    return handle;
-}
+/* mt32-pi: fluid_file_open provided externally */
+
 
 fluid_long_long_t fluid_file_tell(FILE* f)
 {
